src/basal_mass_balance/BMB_prescribed.f90

```diff
@@ -0,0 +1,221 @@
+MODULE BMB_prescribed
+
+  ! Prescribed BMB forcing
+
+! ===== Preamble =====
+! ====================
+
+  USE precisions                                             , ONLY: dp
+  USE mpi_basic                                              , ONLY: par, sync
+  USE control_resources_and_error_messaging                  , ONLY: crash, warning, happy, init_routine, finalise_routine, colour_string
+  USE model_configuration                                    , ONLY: C
+  USE parameters
+  USE mesh_types                                             , ONLY: type_mesh
+  USE ice_model_types                                        , ONLY: type_ice_model
+  USE climate_model_types                                    , ONLY: type_climate_model
+  USE BMB_model_types                                        , ONLY: type_BMB_model
+  USE netcdf_input                                           , ONLY: read_field_from_file_2D
+
+  IMPLICIT NONE
+
+CONTAINS
+
+! ===== Main routines =====
+! =========================
+
+  SUBROUTINE run_BMB_model_prescribed( mesh, ice, BMB, region_name, time)
+    ! Calculate the basal mass balance
+    !
+    ! Prescribed BMB forcing
+
+    IMPLICIT NONE
+
+    ! In/output variables:
+    TYPE(type_mesh),                        INTENT(IN)    :: mesh
+    TYPE(type_ice_model),                   INTENT(IN)    :: ice
+    TYPE(type_BMB_model),                   INTENT(INOUT) :: BMB
+    CHARACTER(LEN=3),                       INTENT(IN)    :: region_name
+    REAL(dp),                               INTENT(IN)    :: time
+
+    ! Local variables:
+    CHARACTER(LEN=256), PARAMETER                         :: routine_name = 'run_BMB_model_prescribed'
+    REAL(dp)                                              :: dummy_dp
+    CHARACTER                                             :: dummy_char
+    CHARACTER(LEN=256)                                    :: choice_BMB_prescribed
+
+    ! Add routine to path
+    CALL init_routine( routine_name)
+
+    ! To prevent compiler warnings
```

Comment:
We're going to figure out a way to remove this crap, modern compilers should be able to see that a variable is not used and just compile without it. Must be a flag for this.

Reply:
Sounds very good. For now as a Fortran amateur, I'll just copy the structure that's used throughout the code